src/smk/layer/layer-vector.js

```diff
@@ -21,54 +21,82 @@ include.module( 'layer.layer-vector-js', [ 'layer.layer-js' ], function () {
         var cv = $( '<canvas width="' + width * mult + '" height="' + height + '">' ).get( 0 )
         var ctx = cv.getContext( '2d' )
 
-        var styles = [].concat( self.config.style )
+        const legendData = [];
+        legendData.push({
+            title: self.config.legend.title || self.config.title,
```

Comment:
I'm not sure if `legend` is required configuration, but you might want some defensive coding here to protect against trying to access the `title` property on something undefined.
eg. `title: (self.config.legend && self.config.legend.title) || self.config.title,`

Reply:
Done.